pyprof/prof/reduction.py

```diff
@@ -93,11 +90,7 @@ def __init__(self, d):
         assert (op == "sum")
         assert (len(args) >= 1)
 
-        #Get input
-        if (args[0]['name'] == ""):
-            i = args[0]
-        else:
-            i = list(filter(lambda x: x['name'] == "input", args))[0]
+        i = args[0]
```

Comment:
Same as above. We cannot assume the arguments appear in the order specified by the API documentation.

Reply:
ditto

There needs to be better testing in this repo else it will be impossible to submit changes

Reply:
The source of the problem is not testing (we can of course improve on that) but the fact that the `name` keyword which has a  well thought out intent and purpose was overloaded and abused without consulting.